packages/@n8n/nodes-langchain/nodes/chains/SentimentAnalysis/SentimentAnalysis.node.ts

```diff
@@ -0,0 +1,230 @@
+							rows: 2,
+						},
+					},
+					{
+						displayName: 'System Prompt Template',
+						name: 'systemPromptTemplate',
+						type: 'string',
+						default: DEFAULT_SYSTEM_PROMPT_TEMPLATE,
+						description: 'String to use directly as the system prompt template',
+						typeOptions: {
+							rows: 6,
+						},
+					},
+					{
+						displayName: 'Include Detailed Results',
+						name: 'includeDetailedResults',
+						type: 'boolean',
+						default: false,
+						description:
+							'Whether to include sentiment strength and confidence scores in the output',
+					},
+					{
+						displayName: 'Enable Auto-Fixing',
+						name: 'enableAutoFixing',
+						type: 'boolean',
+						default: true,
+						description: 'Whether to enable auto-fixing for the output parser',
+					},
+				],
+			},
+		],
+	};
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const items = this.getInputData();
+
+		const llm = (await this.getInputConnectionData(
+			NodeConnectionType.AiLanguageModel,
+			0,
+		)) as BaseLanguageModel;
+
+		const returnData: INodeExecutionData[][] = [];
+
+		for (let itemIdx = 0; itemIdx < items.length; itemIdx++) {
+			const sentimentCategories = this.getNodeParameter(
+				'options.categories',
+				itemIdx,
+				DEFAULT_CATEGORIES,
+			) as string;
+
+			const categories = sentimentCategories.split(',').map((cat) => cat.trim());
+
+			// Initialize returnData with empty arrays for each category
+			if (returnData.length === 0) {
+				returnData.push(
+					...Array(categories.length)
+						.fill([])
+						.map(() => []),
+				);
+			}
+
+			const options = this.getNodeParameter('options', itemIdx, {}) as {
+				systemPromptTemplate?: string;
+				includeDetailedResults?: boolean;
+				enableAutoFixing?: boolean;
+			};
+
+			const schema = z.object({
+				sentiment: z.enum(categories as [string, ...string[]]),
+				strength: z
+					.number()
+					.min(0)
+					.max(1)
+					.describe('Strength score for sentiment in relation to the category'),
+				confidence: z.number().min(0).max(1),
+			});
+
+			const structuredParser = StructuredOutputParser.fromZodSchema(schema);
+
+			const parser = options.enableAutoFixing
+				? OutputFixingParser.fromLLM(llm, structuredParser)
+				: structuredParser;
+
+			const systemPromptTemplate = SystemMessagePromptTemplate.fromTemplate(
+				`${options.systemPromptTemplate ?? DEFAULT_SYSTEM_PROMPT_TEMPLATE}
+	{format_instructions}`,
+			);
+
+			const input = this.getNodeParameter('inputText', itemIdx) as string;
+			const inputPrompt = new HumanMessage(input);
+			const messages = [
+				await systemPromptTemplate.format({
+					categories: sentimentCategories,
+					format_instructions: parser.getFormatInstructions(),
+				}),
+				inputPrompt,
+			];
+
+			const prompt = ChatPromptTemplate.fromMessages(messages);
+			const chain = prompt.pipe(llm).pipe(parser).withConfig(getTracingConfig(this));
+
+			const output = await chain.invoke(messages);
```

Comment:
We may want to wrap this into try..catch. In case LLM returns something not according to zod schema, the structured output parser will throw an exception.

Reply:
True. Wrapped it in try/catch and added handling for `continueOnFail`